### PR TITLE
Rust SDK: Reserved

### PR DESCRIPTION
**Pull request by markmandel**
Implements Reserved, as well as associated tests and examples updated.

Completes Rust work on #927

### CODE REVIEW

**Comment by roberthbailey** [comment]
Who would be a good reviewer for this?

**Comment by markmandel** [comment]
@thara are you able to take a look at my Rust please?

**Comment by thara** [comment]
@markmandel LGTM 👍

File: site/content/en/docs/Guides/Client SDKs/rust.md [hunk 1]
```diff
@@ -52,6 +52,13 @@ To mark the [game session as ready]({{< relref "_index.md#ready" >}}) call `sdk.
 sdk.ready()?;
 ```
 
+
+{{% feature publishVersion-="1.0.0" %}}To mark that the [reserve the game session]({{< relref "_index.md#reserve-seconds" >}}) for a duration call `sdk.reserve(duration)`.
```
Comment (roberthbailey) [inline site/content/en/docs/Guides/Client SDKs/rust.md]: This doesn't read very well:

> To mark that the reserve the game session for a duration call sdk.reserve(duration).

Maybe something like "To mark the game server as reserved for a period of time, call reserve(duration)." (this also matches the nodejs documentation).

Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/rust.md]: I don't even know what I was typing on this one. Lemme fix this now, better to have it for 1.0

Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/rust.md]: Done!

**Pull request merged**